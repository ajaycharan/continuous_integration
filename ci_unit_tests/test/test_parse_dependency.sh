#!/bin/bash -e

for dependency_w_branch in "git@test.test/simple" "git@test.test/simpleWithBranch;test_branch" "git@test.test/just_sha1;b66dda2" "git@test.test/withGitDescirbeLikeRevision;fix/moveRawAccess-429-gb66dda2"; do
  echo -e "********** Testing : $dependency_w_branch ************"

  echo "DependencySpec: $dependency_w_branch"
  
  source ../../modules/parse_dependency.sh

  echo "depth_args: $depth_args"
  echo "clone_args: $clone_args"
done
