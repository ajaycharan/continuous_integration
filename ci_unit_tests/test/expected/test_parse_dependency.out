********** Testing : git@test.test/simple ************
DependencySpec: git@test.test/simple
Dependency: git@test.test/simple
Revision: origin/master
Depth: 1
Foldername: simple
depth_args: --depth 1
clone_args: -b master --single-branch
********** Testing : git@test.test/simpleWithBranch;test_branch ************
DependencySpec: git@test.test/simpleWithBranch;test_branch
Dependency: git@test.test/simpleWithBranch
Revision: origin/test_branch
Depth: 1
Foldername: simpleWithBranch
depth_args: --depth 1
clone_args: -b test_branch --single-branch
********** Testing : git@test.test/just_sha1;b66dda2 ************
DependencySpec: git@test.test/just_sha1;b66dda2
Dependency: git@test.test/just_sha1
Revision: b66dda2
Foldername: just_sha1
depth_args: 
clone_args: 
********** Testing : git@test.test/withGitDescirbeLikeRevision;fix/moveRawAccess-429-gb66dda2 ************
DependencySpec: git@test.test/withGitDescirbeLikeRevision;fix/moveRawAccess-429-gb66dda2
Dependency: git@test.test/withGitDescirbeLikeRevision
Revision: b66dda2
Foldername: withGitDescirbeLikeRevision
depth_args: 
clone_args: 
